src/init.cpp

```diff
@@ -591,7 +591,20 @@ void ThreadImport(std::vector<boost::filesystem::path> vImportFiles)
     // -reindex
     if (fReindex) {
         CImportingNow imp;
+        nSizeReindexed = 0;  // will be modified inside LoadExternalBlockFile
+        // Find the summary size of all block files first
         int nFile = 0;
+        size_t fullSize = 0;
+        while (true) {
+            CDiskBlockPos pos(nFile, 0);
+            boost::filesystem::path blkFile = GetBlockPosFilename(pos, "blk");
+            if (!boost::filesystem::exists(blkFile))
+                break; // No block files left to reindex
+            nFile++;
+            fullSize += boost::filesystem::file_size(blkFile);
+        }
+        nFullSizeToReindex = fullSize;
```

Comment:
I missed a potential divide-by-zero. If we have a zero-length file as the first block file, then https://github.com/zcash/zcash/pull/4368/files#diff-9bee7abc3e254b85a1ce1ed330601054R291 can divide by zero.